qcengine/programs/dftd3.py

```diff
@@ -62,326 +61,224 @@ def get_version(self) -> str:
+            calcinfo.append(qcel.Datum('DISPERSION CORRECTION ENERGY', 'Eh', ene))
+            calcinfo.append(qcel.Datum('2-BODY DISPERSION CORRECTION ENERGY', 'Eh', ene))
             if qcvkey:
-                calcinfo.append(qcel.Datum(f'{qcvkey} DISPERSION CORRECTION GRADIENT', 'Eh/a0', fullgrad))
-
-    calcinfo1 = {info.label: info for info in calcinfo}
-    text += qcel.datum.print_variables(calcinfo1)
-    calcinfo = {info.label: info.data for info in calcinfo}
-    calcinfo = qcel.util.unnp(calcinfo, flat=True)
-
-    # NEW WAY
-    #module_vars = {}
-    #module_vars['DISPERSION CORRECTION ENERGY'] =  ene
-    #module_vars['{} DISPERSION CORRECTION ENERGY'.format(qcvkey)] =  ene
-    #if jobrec['driver'] == 'gradient':
-    #    module_vars['DISPERSION CORRECTION GRADIENT'] = fullgrad
-    #    module_vars['{} DISPERSION CORRECTION GRADIENT'.format(qcvkey)] = fullgrad
-    #
-    #module_vars = PreservingDict(module_vars)
-    #qcvars.build_out(module_vars)
-    #calcinfo = qcvars.certify(module_vars)
-    #text += print_variables(calcinfo)
-
-    jobrec['stdout'] = text
-    jobrec['extras']['qcvars'] = calcinfo
-
-    prov = {}
-    prov['creator'] = 'dftd3'
-    prov['routine'] = sys._getframe().f_code.co_name
-    prov['version'] = version
-    jobrec['provenance'] = prov
-
-    return jobrec
-
-
-def dftd3_coeff_formatter(dashlvl, dashcoeff):
+                calcinfo.append(qcel.Datum(f'{qcvkey} DISPERSION CORRECTION ENERGY', 'Eh', ene))
+
+            if input_model.driver == 'gradient':
+                calcinfo.append(qcel.Datum('CURRENT GRADIENT', 'Eh/a0', fullgrad))
+                calcinfo.append(qcel.Datum('DISPERSION CORRECTION GRADIENT', 'Eh/a0', fullgrad))
+                calcinfo.append(qcel.Datum('2-BODY DISPERSION CORRECTION GRADIENT', 'Eh/a0', fullgrad))
+                if qcvkey:
+                    calcinfo.append(qcel.Datum(f'{qcvkey} DISPERSION CORRECTION GRADIENT', 'Eh/a0', fullgrad))
+
+        #LOGtext += qcel.datum.print_variables({info.label: info for info in calcinfo})
+        calcinfo = {info.label: info.data for info in calcinfo}
+        #calcinfo = qcel.util.unnp(calcinfo, flat=True)
+
+        # got to even out who needs plump/flat/Decimal/float/ndarray/list
+        # Decimal --> str preserves precision
+        calcinfo = {
+            k.upper(): str(v) if isinstance(v, Decimal) else v
+            for k, v in qcel.util.unnp(calcinfo, flat=True).items()
+        }
+
+        # jobrec['properties'] = {"return_energy": ene}
+        # jobrec["molecule"]["real"] = list(jobrec["molecule"]["real"])
+
+        retres = calcinfo[f'CURRENT {input_model.driver.upper()}']
+        if isinstance(retres, Decimal):
+            retres = float(retres)
+        elif isinstance(retres, np.ndarray):
+            retres = retres.ravel().tolist()
+
+        output_data = {
+            'extras': {
+                'local_keywords': input_model.extras['info'],
```

Comment:
No restrictions on extras, the present info seems reasonable however.

Reply:
`local_keywords` is sort of "resolved keywords" and is the best record of what the calc did. I don't really care about the name of the key. Not following "No restrictions on extras."

Reply:
nm, I had forgotten I asked the question at the top, so was trying to figure out the point of your comment.